roles/libvirt_manager/tasks/create_networks.yml

```diff
@@ -81,6 +82,112 @@
   loop_control:
     label: "{{ item.name }}"
 
+## Ensure we get dnsmasq DHCP service for all created networks
+# Refreshing network facts will ensure we get the new interfaces.
+# They (usually?) have the same name as the network itself.
+- name: Refresh networking facts on host
+  ansible.builtin.setup:
+    gather_subset:
+      - '!all,!min'
+      - network
+
+- name: Build needed network related content
+  when:
+    - ansible_facts[_name] is defined
+  vars:
+    _name: "{{ item | replace('-', '_') }}"
+    _no_prefix_name: "{{ _name | regex_replace('cifmw[-_]','') }}"
+    _fixed_nets: >-
+      {{
+        cifmw_libvirt_manager_fixed_networks_defaults +
+        cifmw_libvirt_manager_fixed_networks
+      }}
+    _dns_options: |-
+      {% if ansible_facts[_name].ipv4 is defined -%}
+      - "option:dns-server,{{ ansible_facts[_name].ipv4.address }}"
+      {% endif -%}
+      {% if ansible_facts[_name].ipv6 is defined -%}
+      - "option6:dns-server,[{{ ansible_facts[_name].ipv6.address }}]"
```

Comment:
This causes error `'list object' has no attribute 'address'.` - because ansible fact for ipv4 and ipv6 follow different schema - wee need to do something to select the IPv6 entry where `scope == global` - and then pick the address.

```
  "ipv6": [
    {
      "address": "fc00:abcd:abcd:fc00::1",
      "prefix": "64",
      "scope": "global"
    },
    {
      "address": "fe80::5054:ff:fe43:2504",
      "prefix": "64",
      "scope": "link"
    }
  ],
```

```
  "ipv4": {
    "address": "172.31.255.1",
    "broadcast": "172.31.255.255",
    "netmask": "255.255.255.0",
    "network": "172.31.255.0"
  },
```

Reply:
ah, good to know - I'll check that out, couldn't test actual v6 on my end :/.